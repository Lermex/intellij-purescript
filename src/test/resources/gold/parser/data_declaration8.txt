Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('User8')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(proper name)('User8')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(ObjectType)
            PsiElement({)('{')
            PsiElement(whitespace)('\n  ')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('name')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(TypeConstructor)
                    PSASTWrapperElement(Qualified)
                      PsiElement(proper name)('Email')
                      PsiElement(whitespace)(' ')
                PsiElement(->)('->')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('String')
                        PsiElement(whitespace)('\n ')
            PsiElement(})('}')
            PsiElement(whitespace)(' ')
        PsiElement(|)('|')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(proper name)('B')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(ObjectType)
            PsiElement({)('{')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('kind')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(TypeConstructor)
                    PSASTWrapperElement(Qualified)
                      PsiElement(proper name)('String')
                      PsiElement(whitespace)(' ')
            PsiElement(})('}')