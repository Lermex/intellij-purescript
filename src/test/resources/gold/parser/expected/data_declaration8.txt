Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('User8')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          PsiElement(identifier)('a')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('User8')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
  PsiErrorElement:Unexpected {. Expecting Module
    PsiElement({)('{')
    PsiElement(whitespace)('\n  ')
    PsiElement(identifier)('name')
    PsiElement(whitespace)(' ')
    PsiElement(::)('::')
    PsiElement(whitespace)(' ')
    PsiElement(proper name)('Email')
    PsiElement(whitespace)(' ')
    PsiElement(->)('->')
    PsiElement(whitespace)(' ')
    PsiElement(proper name)('String')
    PsiElement(whitespace)('\n')
    PsiElement(})('}')
    PsiElement(whitespace)(' ')
    PsiElement(|)('|')
    PsiElement(whitespace)(' ')
    PsiElement(proper name)('B')
    PsiElement(whitespace)(' ')
    PsiElement(identifier)('a')
    PsiElement(whitespace)(' ')
    PsiElement({)('{')
    PsiElement(whitespace)(' ')
    PsiElement(identifier)('kind')
    PsiElement(whitespace)(' ')
    PsiElement(::)('::')
    PsiElement(whitespace)(' ')
    PsiElement(proper name)('String')
    PsiElement(whitespace)(' ')
    PsiElement(})('}')